torch/_inductor/lowering.py

```diff
@@ -1019,6 +1019,32 @@ def diagonal_scatter(input, src, offset: int = 0, dim1: int = 0, dim2: int = 1):
     return output
 
 
+@register_lowering(aten.take, type_promotion_kind=None)
+def take(x, index):
+    assert isinstance(x, TensorBox)
+    assert isinstance(index, TensorBox)
+    assert "int" in str(index.get_dtype())
+
+    flat = view(x, [sympy_product(x.get_size())])
+    size = flat.get_size()
+    flat_loader = flat.make_loader()
+    index_loader = index.make_loader()
+
+    def fn(idx):
+        idx = list(idx)
+        idx = index_loader(idx)
+        # deal with negative indices
+        idx = ops.where(ops.lt(idx, 0), idx + ops.constant(size[0], torch.int64), idx)
```

Comment:
Negative indices are dealt with later on, no need to add this here.

Reply:
Ah, right, negative indices should fail on CPU, but not on GPU. We can xfail those for now. Those will work once https://github.com/pytorch/pytorch/pull/108690 is merged.